packages/react-components/src/components/TextField/TextField.tsx

```diff
@@ -0,0 +1,92 @@
+import {
+  TextField as ReactAriaTextField,
+  TextFieldProps as ReactAriaTextFieldProps,
+  Input,
+  Label,
+  FieldError,
+  Text,
+  ValidationResult,
+} from "react-aria-components";
+
+import "./TextField.css";
+
+export interface TextFieldProps extends ReactAriaTextFieldProps {
+  /* Sets size of text input field */
+  size?: "medium" | "small";
+  /* Sets text label above text input field */
+  label?: string;
+  /* Sets optional description text below text input field */
+  description?: string;
+  /* Used for data validation and error handling */
+  errorMessage?: string | ((validation: ValidationResult) => string);
+  /* Icon slot to left of text input field */
+  iconLeft?: React.ReactElement;
+  /* Icon slot to right of text input field */
+  iconRight?: React.ReactElement;
+}
+
+/* Icon displayed when input is in invalid state */
+const iconError = (
```

Comment:
This works here for now. When we have another instance of this icon, we can move the SVG into a common directory as a React component and parameterize the `fill` as a prop.

Reply:
Relevant, from #400: https://github.com/bcgov/design-system/pull/400#issuecomment-2229182789

Reply:
We can update it over there when I get to that one :+1: :+1: